jetty-core/jetty-server/src/test/java/org/eclipse/jetty/server/RequestTest.java

```diff
@@ -100,6 +100,63 @@ public void testEncodedPath() throws Exception
         assertEquals(HttpStatus.BAD_REQUEST_400, response.getStatus());
     }
 
+    public static Stream<Arguments> getUriTests()
+    {
+        return Stream.of(
+            Arguments.of(UriCompliance.DEFAULT, "/", 200, "local"),
+            Arguments.of(UriCompliance.DEFAULT, "https://local/", 200, "local"),
+            Arguments.of(UriCompliance.DEFAULT, "https://other/", 400, "Authority!=Host"),
+            Arguments.of(UriCompliance.DEFAULT, "https://user@local/", 400, "Deprecated User Info"),
+            Arguments.of(UriCompliance.LEGACY, "https://user@local/", 200, "local"),
+            Arguments.of(UriCompliance.DEFAULT, "/%2F/", 400, "Ambiguous URI path separator"),
+            Arguments.of(UriCompliance.UNSAFE, "/%2F/", 200, "local")
```

Comment:
Can we split the good and bad tests apart.

Also, can we use a hostname in the URI that is different than the one used in the `Host: local` header to make sure understand which host it is using?

Reply:
> Can we split the good and bad tests apart.

I  did that, but it only saves a couple of lines and requires 40 lines to be duplicated, so a net loss.   It also makes it hard to see the difference between the 200 and 400 tests, which is good to see exactly what we are testing.  So I prefer it as a single test.

> Also, can we use a hostname in the URI that is different than the one used in the `Host: local` header to make sure understand which host it is using?

I added a bunch more tests, including ones that allow HttpCompliance with different authorities.